### PR TITLE
add reference state to dycore equation documentation

### LINKED ISSUE #1435: Add reference state to the documentation of dycore equations

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

Add reference state to the dycore equation documentation so that it is consistent with the code. Closes #1435 

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: docs/src/equations.md [hunk 1]
```diff
@@ -144,15 +147,15 @@ Similarly for vertical velocity
...
 ```
 The ``(\nabla_v \times \boldsymbol{u}_h + \nabla_h \times \boldsymbol{u}_v) \times \boldsymbol{u}^h`` term is discretized as
 ```math
 (C^f_v[\boldsymbol{u}_h] + C_h[\boldsymbol{u}_v]) \times I^f(\boldsymbol{u}^h) ,
 ```
-and the ``\frac{1}{\rho} \nabla_v p + \nabla_v(\Phi + K)`` term as
+and the ``\frac{1}{\rho} \nabla_v (p - p_h) - \frac{\rho - \rho_h}{\rho} \nabla_v \Phi - \nabla_v K`` term as
 ```math
-\frac{1}{I^f(\rho)} G^f_v[p] + G^f[K + \Phi] ,
+\frac{1}{I^f(\rho)} G^f_v[p - p_h] - frac{I^f(\rho - \rho_h)}{I^f(\rho)} G^f_v[\Phi] - G^f_v[K] ,
```
Comment (Zhengyu-Huang) [inline docs/src/equations.md:158]: \frac

**Review by Zhengyu-Huang — APPROVED** [review]
Thanks for your great work!

File: docs/src/equations.md [hunk 2]
```diff
@@ -65,13 +65,16 @@ We make use of the following operators
 * ``\bar{\boldsymbol{u}} = \boldsymbol{u}_h + I_{c}(\boldsymbol{u}_v)``
 
 * ``\Phi = g z`` is the geopotential, where ``g`` is the gravitational acceleration rate and ``z`` is altitude above the mean sea level.
+* ``\boldsymbol{b} = - (\rho - \rho_h) / \rho \nabla \Phi`` is the reduced gravitational accleration
```
Comment (simonbyrne) [inline docs/src/equations.md:68]:
```suggestion
* ``\boldsymbol{b} = - (\rho - \rho_h) / \rho \nabla \Phi`` is the reduced gravitational acceleration
```

File: docs/src/equations.md [hunk 3]
```diff
@@ -65,13 +65,16 @@ We make use of the following operators
...
+* ``\rho_h`` is the reference state density
 * ``K = \tfrac{1}{2} \|\boldsymbol{u}\|^2 `` is the specific kinetic energy (J/kg), reconstructed at cell centers by
   ```math
   K = \tfrac{1}{2} (\boldsymbol{u}_{h} \cdot \boldsymbol{u}_{h} + 2 \boldsymbol{u}_{h} \cdot \boldsymbol{u}_{v} + I_{c}(\boldsymbol{u}_{v} \cdot \boldsymbol{u}_{v})),
   ```
   where ``\boldsymbol{u}_{h}`` is defined on cell-centers, ``\boldsymbol{u}_{v}`` is defined on cell-faces, and ``I_{c}`` is an interpolation operation using covariant components.  
 
 * ``p`` is air pressure, derived from the thermodynamic state, reconstructed at cell centers.
+* ``p_h`` is the reference state pressure. It is related to the reference state density by hydrostatic balance: ``\Delta p_h = - \rho_h \nabla \Phi``.
```
Comment (simonbyrne) [inline docs/src/equations.md:77]: what are `\Delta` and `\nabla` here?

File: docs/src/equations.md (hunk 2 above)
Comment (simonbyrne) [reply docs/src/equations.md:68]: Also, what is `\nabla`? How is it defined discretely?

File: docs/src/equations.md [hunk 4]
```diff
@@ -65,13 +65,16 @@ We make use of the following operators
 * ``\bar{\boldsymbol{u}} = \boldsymbol{u}_h + I_{c}(\boldsymbol{u}_v)``
 
 * ``\Phi = g z`` is the geopotential, where ``g`` is the gravitational acceleration rate and ``z`` is altitude above the mean sea level.
+* ``\boldsymbol{b} = - (\rho - \rho_h) / \rho \nabla \Phi`` is the reduced gravitational accleration
+* ``\rho_h`` is the reference state density
 * ``K = \tfrac{1}{2} \|\boldsymbol{u}\|^2 `` is the specific kinetic energy (J/kg), reconstructed at cell centers by
   ```math
   K = \tfrac{1}{2} (\boldsymbol{u}_{h} \cdot \boldsymbol{u}_{h} + 2 \boldsymbol{u}_{h} \cdot \boldsymbol{u}_{v} + I_{c}(\boldsymbol{u}_{v} \cdot \boldsymbol{u}_{v})),
   ```
   where ``\boldsymbol{u}_{h}`` is defined on cell-centers, ``\boldsymbol{u}_{v}`` is defined on cell-faces, and ``I_{c}`` is an interpolation operation using covariant components.
```
Comment (simonbyrne) [inline docs/src/equations.md:74]: This needs a bit more:
```suggestion
  ```math
  K = \tfrac{1}{2} (\boldsymbol{u}_{h} \cdot \boldsymbol{u}_{h} + 2 \boldsymbol{u}_{h} \cdot I_{c} (\boldsymbol{u}_{v}) + I_{c}(\boldsymbol{u}_{v} \cdot \boldsymbol{u}_{v})),
  ```
  where ``\boldsymbol{u}_{h}`` is defined on cell-centers, ``\boldsymbol{u}_{v}`` is defined on cell-faces, and ``I_{c} (\boldsymbol{u}_{v})`` is interpolated using covariant components.  
```

File: docs/src/equations.md [hunk 5]
```diff
@@ -65,13 +65,16 @@ We make use of the following operators
 * ``\bar{\boldsymbol{u}} = \boldsymbol{u}_h + I_{c}(\boldsymbol{u}_v)``
 
 * ``\Phi = g z`` is the geopotential, where ``g`` is the gravitational acceleration rate and ``z`` is altitude above the mean sea level.
+* ``\boldsymbol{b} = - (\rho - \rho_h) / \rho \nabla \Phi`` is the reduced gravitational accleration
+* ``\rho_h`` is the reference state density
 * ``K = \tfrac{1}{2} \|\boldsymbol{u}\|^2 `` is the specific kinetic energy (J/kg), reconstructed at cell centers by
   ```math
   K = \tfrac{1}{2} (\boldsymbol{u}_{h} \cdot \boldsymbol{u}_{h} + 2 \boldsymbol{u}_{h} \cdot \boldsymbol{u}_{v} + I_{c}(\boldsymbol{u}_{v} \cdot \boldsymbol{u}_{v})),
```
Comment (simonbyrne) [inline docs/src/equations.md:72]:
```suggestion
K = \tfrac{1}{2} (\boldsymbol{u}_{h} \cdot \boldsymbol{u}_{h} + 2 \boldsymbol{u}_{h} \cdot I_{c} (\boldsymbol{u}_{v}) + I_{c}(\boldsymbol{u}_{v} \cdot \boldsymbol{u}_{v})),
```

File: docs/src/equations.md (hunk 4 above)
Comment (simonbyrne) [inline docs/src/equations.md:74]:
```suggestion
  where ``\boldsymbol{u}_{h}`` is defined on cell-centers, ``\boldsymbol{u}_{v}`` is defined on cell-faces, and ``I_{c} (\boldsymbol{u}_{v})`` is interpolated using covariant components.  
```

File: docs/src/equations.md (hunk 1 above)
Comment (simonbyrne) [inline docs/src/equations.md:158]:
```suggestion
\frac{1}{I^f(\rho)} G^f_v[p - p_h] - \frac{I^f(\rho - \rho_h)}{I^f(\rho)} G^f_v[\Phi] - G^f_v[K] ,
```

File: docs/src/equations.md (hunk 3 above)
Comment (szy21) [reply docs/src/equations.md:77]: Ah, they should be both \nabla. And it is the gradient.

File: docs/src/equations.md (hunk 2 above)
Comment (szy21) [reply docs/src/equations.md:68]: The discretization is the same as the current momentum equation. I actually didn’t use `b` in the vertical momentum equation. Maybe we don’t need to define `b` here?

File: docs/src/equations.md (hunk 3 above)
Comment (szy21) [reply docs/src/equations.md:77]: And here p_h and \rho_h are actually in analytical balance. I will add a note on this.

**Comment by simonbyrne** [comment]
Why the `_h` subscript? We use `_h` elsewhere for "horizontal components", I was wondering if it was confusing.

**Comment by szy21** [comment]
> Why the `_h` subscript? We use `_h` elsewhere for "horizontal components", I was wondering if it was confusing.

I used Tapio’s notation. I’ll change it to `_ref` then.

File: docs/src/equations.md [hunk 6]
```diff
@@ -65,13 +65,16 @@ We make use of the following operators
 * ``\bar{\boldsymbol{u}} = \boldsymbol{u}_h + I_{c}(\boldsymbol{u}_v)``
 
 * ``\Phi = g z`` is the geopotential, where ``g`` is the gravitational acceleration rate and ``z`` is altitude above the mean sea level.
+* ``\boldsymbol{b} = - (\rho - \rho_h) / \rho \nabla \Phi`` is the reduced gravitational acceleration
```
Comment (simonbyrne) [inline docs/src/equations.md:68]: this is a little confusing: is `nabla \Phi` in the numerator or denominator? Maybe use `\frac` and make it a ```math block

Comment (szy21) [reply docs/src/equations.md]: Done.

File: docs/src/equations.md [hunk 7]
```diff
@@ -65,16 +65,19 @@ We make use of the following operators
 * ``\bar{\boldsymbol{u}} = \boldsymbol{u}_h + I_{c}(\boldsymbol{u}_v)``
 
 * ``\Phi = g z`` is the geopotential, where ``g`` is the gravitational acceleration rate and ``z`` is altitude above the mean sea level.
-* ``\boldsymbol{b} = - (\rho - \rho_h) / \rho \nabla \Phi`` is the reduced gravitational acceleration
-* ``\rho_h`` is the reference state density
+* ``\boldsymbol{b}`` is the reduced gravitational acceleration
+   ```math
+   \boldsymbol{b} = - \frac{\rho - \rho_ref}{\rho} \nabla \Phi
+   ```
+* ``\rho_ref`` is the reference state density
```
Comment (simonbyrne) [inline docs/src/equations.md:72]:
```suggestion
* ``\rho_{\text{ref}}`` is the reference state density
```

Comment (szy21) [reply docs/src/equations.md:72]: Ah, right, I'll change all of them.

Comment (simonbyrne) [reply docs/src/equations.md:72]: and elsewhere

File: docs/src/equations.md [hunk 8]
```diff
@@ -65,13 +65,19 @@ We make use of the following operators
 * ``\bar{\boldsymbol{u}} = \boldsymbol{u}_h + I_{c}(\boldsymbol{u}_v)``
 
 * ``\Phi = g z`` is the geopotential, where ``g`` is the gravitational acceleration rate and ``z`` is altitude above the mean sea level.
+* ``\boldsymbol{b}`` is the reduced gravitational acceleration
+   ```math
```
Comment (simonbyrne) [inline docs/src/equations.md:69]: hmm, may need to unindent by 1?
```suggestion
  ```math
```

Comment (simonbyrne) [inline docs/src/equations.md:69]: may need to unindent by 1 space
```suggestion
  ```math
```

**Comment by szy21** [comment]
@simonbyrne I think it's finally correct now

**Pull request merged**